### PR TITLE
Add iOS Target

### LINKED ISSUE #19: iOS target?
I tried to use Either in a combined Mac/iOS library I'm developing, but looks like this project is only configured for Mac targets. Any reason not to add iOS as well?

### PR DESCRIPTION
**Pull request by gfontenot**
This adds an iOS target for Either.

Fixes #19

### CODE REVIEW

File: Either.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -27,20 +37,29 @@
...
 		D4E7A7F01A2CCB800080A4DF /* EitherTypeTests.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = EitherTypeTests.swift; sourceTree = "<group>"; };
+		F8BB81FA1A939EA0001AA352 /* Either.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Either.framework; sourceTree = BUILT_PRODUCTS_DIR; };
+		F8BB82041A939EA0001AA352 /* Either-iOSTests.xctest */ = {isa = PBXFileReference; explicitFileType = wrapper.cfbundle; includeInIndex = 0; path = "Either-iOSTests.xctest"; sourceTree = BUILT_PRODUCTS_DIR; };
+		F8BB821B1A939FB8001AA352 /* Prelude.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Prelude.framework; path = "../../../../Library/Developer/Xcode/DerivedData/Either-eaadslcryzlcpaaxuskiptjrhbhe/Build/Products/Debug/Prelude.framework"; sourceTree = "<group>"; };
+		F8BB821D1A939FBB001AA352 /* Box.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Box.framework; path = "../../../../Library/Developer/Xcode/DerivedData/Either-eaadslcryzlcpaaxuskiptjrhbhe/Build/Products/Debug/Box.framework"; sourceTree = "<group>"; };
```
Comment (robrix) [inline Either.xcodeproj/project.pbxproj]: If you show the File Inspector pane and then select the references to these in the File Navigator, you can use the “Location” pop-up to set these to be relative to the built products dir. Could you do that please?

File: Either.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -345,11 +474,15 @@
 				DYLIB_COMPATIBILITY_VERSION = 1;
 				DYLIB_CURRENT_VERSION = 1;
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)",
+					"$(USER_LIBRARY_DIR)/Developer/Xcode/DerivedData/Either-eaadslcryzlcpaaxuskiptjrhbhe/Build/Products/Debug",
+				);
```
Comment (robrix) [inline Either.xcodeproj/project.pbxproj]: Could you please delete this build setting?

File: Either.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -380,6 +517,7 @@
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(DEVELOPER_FRAMEWORKS_DIR)",
 					"$(inherited)",
+					"$(USER_LIBRARY_DIR)/Developer/Xcode/DerivedData/Either-eaadslcryzlcpaaxuskiptjrhbhe/Build/Products/Debug",
```
Comment (robrix) [inline Either.xcodeproj/project.pbxproj]: Likewise this one (on the tests).

File: Either.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -405,6 +544,102 @@
...
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				DEBUG_INFORMATION_FORMAT = dwarf;
+				DEFINES_MODULE = YES;
+				DYLIB_COMPATIBILITY_VERSION = 1;
+				DYLIB_CURRENT_VERSION = 1;
+				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)",
+					"$(USER_LIBRARY_DIR)/Developer/Xcode/DerivedData/Either-eaadslcryzlcpaaxuskiptjrhbhe/Build/Products/Debug",
+				);
```
Comment (robrix) [inline Either.xcodeproj/project.pbxproj]: Ditto for the iOS framework and tests.

**Comment by robrix** [comment]
Looks good; just a couple of notes.

**Comment by robrix** [comment]
I’m going to merge and catch the above separately.

**Comment by gfontenot** [comment]
@robrix Oops, sorry about those issues. Thanks for taking care of it!

**Comment by robrix** [comment]
No worries, that’s the magic of git :sparkles: :grin:

**Pull request merged**